src/main/interface/cli.c

```diff
@@ -3310,6 +3311,27 @@ static void resourceCheck(uint8_t resourceIndex, uint8_t index, ioTag_t newTag)
     }
 }
 
+static bool strToPin(char *pch, ioTag_t *tag)
+{
+    if (strcasecmp(pch, "NONE") == 0) {
+        *tag = IO_TAG_NONE;
+        return true;
+    } else {
+        unsigned pin = 0;
+        unsigned port = (*pch >= 'a') ? *pch - 'a' : *pch - 'A';
+
+        if (port < 8) {
```

Comment:
`DEFIO_PORT_USED_COUNT` is count of used ports / max port +1 

Also `io.h` is making sure that only valid tags are possible, maybe invalid tag may cause problems. It is possible to check it dynamically - bitmask in in `ioDefUsedMask`

It is probably good idea to move the checks into `io.c` - `IOCreateTag(portid, pinid)`

Reply:
Agreed. I can tidy that up in another PR.